src/Auth.ts

```diff
@@ -706,6 +718,24 @@ export class Auth {
     });
   }
 
+  private async ensureAccessTokenWithAccessToken(resource: string, logger: Logger, debug: boolean): Promise<AccessToken | null> {
+    const accessToken: AccessToken | undefined = this.connection.accessTokens[resource];
+
+    if (!accessToken) {
+      throw `No token found for resource ${resource}.`;
+    }
+
+    if (this.accessTokenExpired(accessToken)) {
```

Comment:
We should handle this centrally for all tokens rather than for each auth type separately. The logic is the same and having it done early allows us to speed up auth flow and avoid repetition

Reply:
Well, yes, but in the case of this new authType we want slightly different behavior: I want to `throw` if it's expired for example. Do you have suggestions on how to do it differently?